ip_validation/ip_validation.rb

```diff
@@ -0,0 +1,9 @@
+require 'ipaddr'
+
+def ip_validation(str)
```

Comment:
1. in ruby we could name the methods with `?` at the end to signify that this method returns a boolean.
  `ip_validation(str)` → `ipv4?(str)`
2. we could drop `begin` inside the method and just apply `rescue` to the block.

so the final solution could look like this: 

```ruby
def ipv4?(str)
  IPAddr.new(str).ipv4?
rescue IPAddr::InvalidAddressError
  false
end
```

Reply:
Refactored